asciidoc/modules/con_mg_improvements-in-vertx-http-client.adoc

```diff
@@ -0,0 +1,203 @@
+}).netSocket(ar -> {
+  if (ar.succeeded()) {
+   // Got a response with a 200 status code
+   NetSocket so = ar.result();
+   // Go for tunneling
+  }
+}).end();
+----
+
+== New `send()` method in `HttpClient` class
+
+A new `send()` method is available in the `HttpClient` class. This method enables you to send requests in a generic way.
+
+The following code shows how to send a request in {VertX} {4}.
+
+----
+Future<HttpClientResponse> f1 = client.send(HttpMethod.GET, 8080, "localhost", "/uri", HttpHeaders.set("foo", "bar"));
+----
+
+== HttpHeaders is an interface and contains `MultiMap` methods
+
+In {VertX} {4}, `HttpHeaders` is an interface. In earlier releases of {VertX},  `HttpHeaders` was a class.
+
+The following new `MultiMap` methods have been added in the `HttpHeaders` interface. Use these methods to create `MultiMap` instances.
+
+* `MultiMap.headers()`
+* `MultiMap.set(CharSequence name, CharSequence value)`
+* `MultiMap.set(String name, String value)`
+
+The following example shows how `MultiMap` instances were created in {VertX} {3x} releases.
+
+----
+MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+----
+
+The following examples show how to create `MultiMap` instances in {VertX} {4}.
+
+----
+MultiMap headers = HttpHeaders.headers();
+----
+----
+MultiMap headers = HttpHeaders.set("content-type", "application.data");
+----
+
+== `CaseInsensitiveHeaders` class is no longer public
+
+The `CaseInsensitiveHeaders` class is no longer public. Use the `MultiMap.caseInsensitiveMultiMap()` method to create a multi-map implementation with case insensitive keys.
+
+The following example shows how `CaseInsensitiveHeaders` method was used in {VertX} {3x} releases.
+
+----
+CaseInsensitiveHeaders headers = new CaseInsensitiveHeaders();
+----
+
+The following examples show how `MultiMap` method is used in {VertX} {4}.
+
+----
+MultiMap multiMap = MultiMap#caseInsensitiveMultiMap();
+----
+OR
+----
+MultiMap headers = HttpHeaders.headers();
+----
+
+== Checking the version of HTTP running on the server
+
+In earlier releases of {VertX}, the version of HTTP running on a server was checked only if the application explicitly called the `HttpServerRequest.version()` method. If the HTTP version was HTTP/1.x, the method would return the 501 HTTP status, and close the connection.
+
+From {VertX} {4} onward, before a request is sent to the server, the HTTP version on the server is automatically checked by calling the `HttpServerRequest.version()` method. The method returns the HTTP version instead of throwing an exception when an invalid HTTP version is found.
+
+== New methods in request options
+
+In {VertX} {4}, the following new methods are available in the `RequestOptions` class:
+
+* Header
+* FollowRedirects
+* Timeout
+
+The following example shows you how to use these new methods:
+
+[source,java,options="nowrap",subs="attributes+"]
+----
+client.request(HttpMethod.GET, 8080, "example.com", "/resource", ar -> {
```

Comment:
@vietj Adam asked me add the method "HttpMethod.GET" in the first line of code:
client.request(HttpMethod.GET, 8080, "example.com", "/resource", ar -> {

Original note:
https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-enhanced-request-options